genomic_address_service/classes/assign.py

```diff
@@ -153,31 +184,29 @@ def get_threshold_idx(self,dist):
         return 0
 
     def guess_file_type(self,f):
-        file_type = None
-        for t in EXTENSIONS:
-            for e in EXTENSIONS[t]:
-                if e in f:
-                    file_type = t
-                    break
-        return file_type
+        extension = os.path.splitext(f)[1]
+        valid_extensions = list(EXTENSIONS.keys())
+
+        if not extension in valid_extensions:
+            message = f'{f} does not have a valid extension {valid_extensions}'
+            raise Exception(message)
+
+        return EXTENSIONS[extension]
 
-    def read_data(self,f):
+    def read_data(self, f):
         df = pd.DataFrame()
         file_type = self.guess_file_type(f)
-        if file_type == 'text':
+
+        if file_type == TEXT:
             df = pd.read_csv(f, header=0, sep="\t", low_memory=False)
-        elif file_type == 'parquet':
-            df = pd.read_parquet(
-                f,
-                engine='auto',
-                columns=None,
-                storage_options=None,)
+        else:
+            message = f'{f} does not have a valid extension'
```

Comment:
Add info what extension was provided by the user and print out what are valid extensions

Reply:
I agree that this is confusing. The code was already doing what you were suggest, but in the `guess_file_type` part. I've cleaned up the code to this instead:

```python
def check_file_type(self,f):
    extension = os.path.splitext(f)[1]
    valid_extensions = list(EXTENSIONS.keys())

    if not extension in valid_extensions:
        message = f'{f} does not have a valid extension {valid_extensions}'
        raise Exception(message)

    return EXTENSIONS[extension]

def read_data(self, f):
    self.check_file_type(f)
    df = pd.read_csv(f, header=0, sep="\t", low_memory=False)

    return df
````

updated (and more changes) here: 69849c9